### DISCUSSION TITLE
Processing check-in of "big" amounts of products (e.g. merch pick-up)

### DISCUSSION THREAD

**Discussion by c0de-bender** [pretix/pretix#3703] [Q&A] [answered]
Hi,

we have an event where we want to sell merch (e.g. T-shirts, cups, buttons). Let's assume someone books entrance tickets for 6 people, and also:

  - 3 t-shirts size S
  - 3 t-shirts size M
  - 4 t-shirts size L
  - 10 cups
  - 7 red buttons
  - 13 green buttons

Does Pretix (or pretixSCAN) provide a fast way to process this at the merch desk? I wouldn't want to scan 40 QR codes.

My desired solution would basically show me the number of items bought per product variation (basically as written down above). I don't really care if:

  - we'd need to scan 1 QR code and then be presented with this list,
  - or print a paper (spreadsheet) list which Pretix produces for us, stating those numbers in columns, each line identifiable by e.g. order number.

If there's an automated process with Pretix (or pretixSCAN), it would ideally mark the 40 merch products as picked up ("checked in") in the system – so that every booked merch product can be picked up only once.

### Remarks

For this to work, we don't care if the merch needs to be configured as a main or as an add-on product. Either would work for us.

I just realised that handling merch this way does introduce other problems. Like people could not pick up only part of their merch at some point and the rest later (i.e. the person booking the merch would need all the other 5 people, if they want to check if the t-shirt sizes fit). So they could not distribute single item tickets to different people to all pick up their own t-shirts by themselves. We could live with that drawback, though. If a combination of both is possible (single and batch pick-up both possible), that would be even cooler (but pretty advanced).

Note: This question is related to [getting information on add-on products when scanning a main product](https://github.com/pretix/pretix/discussions/3702), but it's not the same.

PS: I cannot imagine no one asked this before, but somehow I didn't find a similar question.

**Comment by raphaelm** [comment, marked as answer]
> Does Pretix (or pretixSCAN) provide a fast way to process this at the merch desk? I wouldn't want to scan 40 QR codes.

Currently not.

> we'd need to scan 1 QR code and then be presented with this list,

This is planned for the semi-near future for add-on products.

> I just realised that handling merch this way does introduce other problems. Like people could not pick up only part of their merch at some point and the rest later (i.e. the person booking the merch would need all the other 5 people, if they want to check if the t-shirt sizes fit)

You have correctly identified why this is a hard area of improvement ;)

**Reply by c0de-bender** [reply]
Thanks for the reply. It's great to hear that the "one QR code including all add-ons" feature is on the roadmap! Whatever "semi-near future" means :smile: I'll be patient and for now we can definitely find a different solution. :)
